backend/functions/src/emailing/functions.ts

```diff
@@ -133,18 +156,41 @@ export const sendMails = async (
       },
       data: JSON.stringify(message),
     })
-    if (response.status === 202 && parseInt(group) > 0) {
-      await updateEmailTimestamp(courseId, group, template)
-        .then(() =>
-          logger.info(
-            `Added timestamp for course ${courseId} for group ${group} with template ${template}`
+
+    if (response.status === 202) {
+      if (parseInt(group) > 0) {
+        await updateEmailTimestamp(courseId, group, template)
+          .then(() =>
+            logger.info(
+              `Added timestamp for course ${courseId} for group ${group} with template ${template}`
+            )
           )
-        )
-        .catch((err) =>
-          logger.error(
-            `Failed to update timestampfor course ${courseId} for group ${group} with template ${template}. Resulted in err: ${err.message} `
+          .catch((err) =>
+            logger.error(
+              `Failed to update timestamp for course ${courseId} for group ${group} with template ${template}. Resulted in err: ${err.message} `
+            )
           )
+      }
+
+      if (typeof indivEmail !== 'undefined') {
+        await updateIndivTimestamp(courseId, indivEmail)
+          .then(() =>
+            logger.info(
+              `Added no match timestamp for course ${courseId} for student with email ${indivEmail}`
+            )
+          )
+          .catch((err) =>
+            logger.error(
+              `Failed to update no match timestamp for course ${courseId} for student with email ${indivEmail}. Resulted in err: ${err.message} `
+            )
+          )
+      }
+
+      if (parseInt(group) < 0 && typeof indivEmail === undefined) {
+        logger.error(
+          ` Invalid group ${group} and invalid email ${indivEmail} for updating timestamps `
```

Comment:
I think this check should be earlier rather than later, so just in case the group and individual email are invalid it doesn't even try to send the emails before figuring out it can't update timestamps. Proactively detecting bad input is important for backend services and in this case we would want to error as early as possible. There should also be a guard for when there's both a group number and individual email supplied, because I think this an either-or scenario (unless that assumption is wrong, but at least it lines up with the frontend)?